File: languages/js/test/e2e/identity_core.test.ts
import { describe, test, expect } from "@jest/globals";
import {
    DidMethod,
    IdentityCoreClient,
    Key,
    PublishIntervalParams,
} from "../../dist/index";
import {
    IdentityClient,
    KeyClient,
    KeyProtectionLevel,
    KeyType,
    ManagedKeyParams,
} from "../../dist";
import { initDevSdk } from "./util";

describe("Identity V2 Tests", () => {
    const drivingLicenseSchemaType = "DrivingLicense";
    const holderDid =
        "did:polygonid:polygon:mumbai:2qGg7TzmcoU4Jg3E86wXp4WJcyGUTuafPZxVRxpYQr";
    const expiration = 4089852142;

    test("test identity core end to end", async () => {
        initDevSdk();

        const identityClient = new IdentityClient();
        const keyClient = new KeyClient();
        const identityCoreClient = new IdentityCoreClient();

        let keyProtection = KeyProtectionLevel.SOFTWARE;
        let keyType = KeyType.Bjj;
        let managedKey = await keyClient.newManagedKey(
            new ManagedKeyParams(keyProtection, keyType)
        );

        let issuerKey = new Key(managedKey);

        let issuer = await identityClient.createIssuer(
            issuerKey,
            PublishIntervalParams.Interval15,
            DidMethod.PolygonIDTest,
            "Bloock Test",
            "bloock description test",
        );
        expect(issuer.did.did.includes("polygonid")).toBeTruthy();

        let schema = await identityClient
            .buildSchema(
                "Driving License",
                drivingLicenseSchemaType,
                "1.0",
                "driving license schema"
            )
            .addIntegerAttribute(
                "License Type",
                "license_type",
                "license type",
                false
            )
            .addDecimalAttribute("Quantity Oil", "quantity_oil", "quantity oil", true)
            .addStringAttribute("Nif", "nif", "nif", true)
            .addBooleanAttribute("Is Spanish", "is_spanish", "is spanish", true)
            .addDateAttribute("Birth Date", "birth_date", "birth date", true)
            .addDateTimeAttribute("Local Hour", "local_hour", "local hour", true)
            .addStringEnumAttribute("Car Type", "car_type", "car type", true, [
                "big",
                "medium",
                "small"
            ])
            .addIntegerEnumAttribute("Car Points", "car_points", "car points", true, [
                1,
                5,
                10
            ])
            .addDecimalEnumAttribute(
                "Precision wheels",
                "precision_wheels",
                "precision whels",
                true,
                [1.1, 1.2, 1.3]
            )
            .build();
        expect(schema.cid).toBeTruthy();

        schema = await identityClient.getSchema(schema.cid);
        expect(schema.cidJsonLd).toBeTruthy();
        expect(schema.json).toBeTruthy();
        expect(schema.schemaType).toBeTruthy();

        const receipt = await identityCoreClient
            .buildCredential(issuer, schema.cid, holderDid, expiration, 0)
            .withIntegerAttribute("license_type", 1)
            .withDecimalAttribute("quantity_oil", 2.25555)
            .withStringAttribute("nif", "54688188M")
            .withBoleanAttribute("is_spanish", true)
            .withDateAttribute("birth_date", new Date(1999, 3, 20))
            .withDateTimeAttribute("local_hour", new Date(Date.now()))
            .withStringAttribute("car_type", "big")
            .withIntegerAttribute("car_points", 5)
            .withDecimalAttribute("precision_wheels", 1.1)
            .build();
        expect(receipt.credentialId).toBeTruthy();
        expect(receipt.credential).toBeTruthy();
        expect(receipt.credentialType).toStrictEqual(drivingLicenseSchemaType);

        let credential = receipt.credential;
        expect(credential.issuer).toStrictEqual(issuer.did.did);
        expect(credential.credentialSchema.type).toStrictEqual("JsonSchema2023");
        expect(credential.type[1]).toStrictEqual(drivingLicenseSchemaType);
    })
})

File: languages/go/client/identity_core_test.go
//go:build e2e

package client

import (
	"testing"
	"time"

	"github.com/bloock/bloock-sdk-go/v2/entity/identity"
	"github.com/bloock/bloock-sdk-go/v2/entity/key"
	"github.com/stretchr/testify/assert"
	"github.com/stretchr/testify/require"
)

func TestIdentityCore(t *testing.T) {
	DrivingLicenseSchemaType := "DrivingLicense"
	holderDid := "did:polygonid:polygon:mumbai:2qGg7TzmcoU4Jg3E86wXp4WJcyGUTuafPZxVRxpYQr"
	expiration := int64(4089852142)

	InitDevSdk()

	t.Run("identity core end to end create credential", func(t *testing.T) {
		identityClient := NewIdentityClient()
		keyClient := NewKeyClient()
		identityCoreClient := NewIdentityCoreClient()

		managedKey, err := keyClient.NewManagedKey(key.ManagedKeyParams{
			KeyType:    key.Bjj,
			Protection: key.KEY_PROTECTION_SOFTWARE,
		})
		require.NoError(t, err)

		issuerKey := key.Key{ManagedKey: &managedKey}

		issuer, err := identityClient.CreateIssuer(issuerKey, identity.Interval60, identity.PolygonIDTest, "SDK Issuer Test Core Client", "sdk issuer test core client", "")
		require.NoError(t, err)

		schema, err := identityClient.BuildSchema("Driving License", DrivingLicenseSchemaType, "1.0", "driving license schema").
			AddIntegerAttribute("License Type", "license_type", "license type", false).
			AddDecimalAttribute("Quantity Oil", "quantity_oil", "quantity oil", true).
			AddStringAttribute("Nif", "nif", "nif", true).
			AddBooleanAttribute("Is Spanish", "is_spanish", "is spanish", true).
			AddDateAttribute("Birth Date", "birth_date", "birth date", true).
			AddDatetimeAttribute("Local Hour", "local_hour", "local hour", true).
			AddStringEnumAttribute("Car Type", "car_type", "car type", true, []string{"big", "medium", "small"}).
			AddIntegerEnumAttribute("Car Points", "car_points", "car points", true, []int64{1, 5, 10}).
			AddDecimalEnumAttribute("Precision wheels", "precision_wheels", "precision wheels", true, []float64{1.10, 1.20, 1.30}).
			Build()
		assert.NoError(t, err)
		assert.NotNil(t, schema.Cid)

		schema, err = identityClient.GetSchema(schema.Cid)
		assert.NoError(t, err)
		assert.NotNil(t, schema.CidJsonLd)
		assert.NotNil(t, schema.Json)
		assert.NotNil(t, schema.SchemaType)

		res, err := identityCoreClient.BuildCredential(issuer, schema.Cid, holderDid, expiration, 0).
			WithIntegerAttribute("license_type", 1).
			WithDecimalAttribute("quantity_oil", 2.25555).
			WithStringAttribute("nif", "54688188M").
			WithBooleanAttribute("is_spanish", true).
			WithDateAttribute("birth_date", time.Date(1999, time.March, 20, 0, 0, 0, 0, time.UTC)).
			WithDatetimeAttribute("local_hour", time.Now()).
			WithStringAttribute("car_type", "big").
			WithIntegerAttribute("car_points", 5).
			WithDecimalAttribute("precision_wheels", 1.10).
			Build()
		assert.NoError(t, err)
		assert.NotNil(t, res.CredentialId)
		assert.NotNil(t, res.Credential)
		assert.Equal(t, DrivingLicenseSchemaType, res.CredentialType)

		credential := res.Credential
		assert.Equal(t, issuer.Did.Did, credential.Issuer)
		assert.Equal(t, "JsonSchema2023", credential.CredentialSchema.Type)
		assert.Equal(t, DrivingLicenseSchemaType, credential.Type[1])
	})
}
